cdap-data-fabric/src/main/java/co/cask/cdap/data2/increment/hbase/IncrementHandlerState.java

```diff
@@ -54,17 +54,19 @@
   public static final int BATCH_UNLIMITED = -1;
 
   public static final Log LOG = LogFactory.getLog(IncrementHandlerState.class);
+  private final String tableName;
+  private final HBaseTableUtil tableUtil;
 
   private TransactionStateCache cache;
   private TimestampOracle timeOracle = new TimestampOracle();
-  private final String tableName;
   private final Configuration conf;
   protected final Set<byte[]> txnlFamilies = Sets.newTreeSet(Bytes.BYTES_COMPARATOR);
   protected Map<byte[], Long> ttlByFamily = Maps.newTreeMap(Bytes.BYTES_COMPARATOR);
 
-  public IncrementHandlerState(Configuration conf, String tableName) {
+  public IncrementHandlerState(Configuration conf, String tableName, HBaseTableUtil tableUtil) {
```

Comment:
I don't think it's a good idea to bring in `HBaseTableUtil` class dependencies into coprocessor. The `HBaseTableUtil` was intended for "client" side usage.

As I mentioned before, you should just have a simple hbase version specific util class that has static methods to encode/decode version specific table name such that both version specific HBase table util and coprocessor can use it.